common/test_filestore_utils.py

```diff
@@ -22,26 +22,63 @@
 # the local_filestore implementation.
 
 
-def test_keyword_args():
-    """Tests that keyword args, and in particular 'parallel' are handled
-    correctly."""
-    filestore_path = 'gs://fake_dir'
-
-    with mock.patch('common.new_process.execute') as mocked_execute:
-        filestore_utils.rm(filestore_path, recursive=True, parallel=True)
-        mocked_execute.assert_called_with(
-            ['gsutil', '-m', 'rm', '-r', filestore_path], expect_zero=True)
-
-    with mock.patch('common.new_process.execute') as mocked_execute:
-        mocked_execute.return_value = new_process.ProcessResult(0, '', '')
-        filestore_utils.ls(filestore_path)
-        mocked_execute.assert_called_with(['gsutil', 'ls', filestore_path],
-                                          expect_zero=True)
-
-    filestore_path2 = filestore_path + '2'
-
-    with mock.patch('common.new_process.execute') as mocked_execute:
-        filestore_utils.cp(filestore_path, filestore_path2, parallel=True)
-        mocked_execute.assert_called_with(
-            ['gsutil', '-m', 'cp', filestore_path, filestore_path2],
-            expect_zero=True)
+class TestLocalFilestore:
+    """Tests for switching on local_filestore."""
+    FILESTORE = '/experiment_data'
+    DIR1 = '/dir1'
+    DIR2 = '/dir2'
+
+    def test_local_filestore_on(self, use_local_filestore):  # pylint: disable=unused-argument
+        """Tests that local_filestore switches on correctly."""
+        with mock.patch('common.new_process.execute') as mocked_execute:
+            filestore_utils.cp(self.DIR1, self.DIR2, recursive=True)
+            assert 'gsutil' not in mocked_execute.call_args_list[0][0][0]
+
+
+class TestGsutil():
+    """Tests for switching on gsutil."""
+    FILESTORE = 'gs://experiment-data'
+    LOCAL_DIR = '/dir'
+    GSUTIL_DIR = 'gs://fake_dir'
+
+    def test_gsutil_on(self, fs, use_gsutil):  # pylint: disable=invalid-name,unused-argument
+        """Tests that gsutil switches on correctly."""
+
+        with mock.patch('common.new_process.execute') as mocked_execute:
+            fs.create_dir(self.LOCAL_DIR)
+            filestore_utils.cp(self.LOCAL_DIR, self.GSUTIL_DIR, recursive=True)
+            mocked_execute.assert_called_with(
+                ['gsutil', 'cp', '-r', self.LOCAL_DIR, self.GSUTIL_DIR],
+                expect_zero=True)
+
+    def test_keyword_args(self, use_gsutil):  # pylint: disable=unused-argument
```

Comment:
I think we need some tests that ensure parallel doesn't affect local_filestore

Reply:
I delete that, since right now, there are 4 related `# pylint: disable=unused-arguments` for `parallel` in their arguments.
I don't think we need to write a unit test for it since the argument is not used at all.

Previously, if we have `local_filestore_command` then I can check whether `parallel` has passed to that function. However, since we deleted that function, it is also awkward to write unit tests.

Reply:
Just check that parallel doesn't affect the command passed to `execute` this is the more important thing to test anyway.